demux/utils/novaseq_samplesheet.py

```diff
@@ -0,0 +1,203 @@
+                        added_dummy_samples.append(add_dummy_sample)
+
+            self.raw_samplesheet.extend(added_dummy_samples)
+
+            return raw_samplesheet
+
+    def remove_unwanted_indexes(self, raw_samplesheet) -> "CreateNovaseqSamplesheet":
+        """ Filter out indexes of unwanted length and single indexes """
+
+        raw_samplesheet = [
+            line for line in raw_samplesheet if self.is_dual_index(line["index"])
+        ]
+        return raw_samplesheet
+
+    def adapt_indexes(self, raw_samplesheet) -> "CreateNovaseqSamplesheet":
+        """Adapts the indexes: pads all indexes so that all indexes have a length equal to the
+        number  of index reads, and takes the reverse complement of index 2 in case of the new
+        novaseq software control version (1.7) in combination with the new reagent kit
+        (version 1.5)"""
+
+        reverse_complement = self.is_reverse_complement()
+
+        for line in raw_samplesheet:
+            index1, index2 = line["index"].split("-")
+            if self.pad and len(index1) == 8:
+                line["index"], line["index2"] = self.pad_and_rc_indexes(
+                    index1, index2, reverse_complement
+                )
+            elif len(index2) == 10:
+                line["index2"] = (
+                    self.reverse_complement(index2) if reverse_complement else index2
+                )
+            else:
+                line["index"], line["index2"] = index1, index2
+
+        return raw_samplesheet
+
+    def pad_and_rc_indexes(
+        self, index1: str, index2: str, reverse_complement: bool
+    ) -> tuple:
+        """ Pads and reverse complements indexes """
+
+        if self.runparameters.index_reads == 8:
+            index2 = self.reverse_complement(index2) if reverse_complement else index2
+        if self.runparameters.index_reads == 10:
+            index1 = index1 + "AT"
+            index2 = (
+                self.reverse_complement("AC" + index2)
+                if reverse_complement
+                else index2 + "AC"
+            )
+
+        return index1, index2
+
+    def reagent_kit_version(self) -> int:
+        """ Derives the reagent kit version from the run parameters """
+
+        parameter_to_version = {"1": 1.0, "3": 1.5}
+
+        return parameter_to_version.get(self.runparameters.reagent_kit_version)
+
+    def construct_samplesheet(self, end="\n", delimiter=COMMA) -> str:
+        """ Construct the sample sheet """
+
+        demux_samplesheet = [delimiter.join(self.header)]
+        raw_samplesheet = self.raw_samplesheet
+        raw_samplesheet = self.add_dummy_indexes(raw_samplesheet)
+        raw_samplesheet = self.remove_unwanted_indexes(raw_samplesheet)
+        raw_samplesheet = self.adapt_indexes(raw_samplesheet)
+        for line in raw_samplesheet:
+            # fix the project content
+            project = self.get_project_name(line["project"])
+            line["project"] = project
```

Comment:
```suggestion
             line["project"] = self.get_project_name(line["project"])
```
Does one need this extra line in python?

Reply:
No, but I'm using `project` gain on line 197: `line["sample_name"] = project`.